File: binding/src/charts/libqbarlegendmarker.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qbarlegendmarker.html
pub const qbarlegendmarker = struct {

    /// New constructs a new QBarLegendMarker object.
    pub fn New(series: ?*anyopaque, barset: ?*anyopaque, legend: ?*anyopaque) QtC.QBarLegendMarker {
        return qtc.QBarLegendMarker_new(@ptrCast(series), @ptrCast(barset), @ptrCast(legend));
    }


    /// New2 constructs a new QBarLegendMarker object.
    pub fn New2(series: ?*anyopaque, barset: ?*anyopaque, legend: ?*anyopaque, parent: ?*anyopaque) QtC.QBarLegendMarker {
        return qtc.QBarLegendMarker_new2(@ptrCast(series), @ptrCast(barset), @ptrCast(legend), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QBarLegendMarker_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbarlegendmarker.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QBarLegendMarker_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QBarLegendMarker_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QBarLegendMarker_QBaseType(@ptrCast(self));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QAbstractBarSeries {
        return qtc.QBarLegendMarker_Series(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSeries(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QAbstractBarSeries) void {
        qtc.QBarLegendMarker_OnSeries(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSeries(self: ?*anyopaque, ) QtC.QAbstractBarSeries {
        return qtc.QBarLegendMarker_QBaseSeries(@ptrCast(self));
    }

    pub fn Barset(self: ?*anyopaque, ) QtC.QBarSet {
        return qtc.QBarLegendMarker_Barset(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QBarLegendMarker_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbarlegendmarker.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QBarLegendMarker_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qbarlegendmarker.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QBarLegendMarker_Delete(@ptrCast(self));
    }
};

File: binding/src/charts/libqbarlegendmarker.cpp
#include <QAbstractBarSeries>
#include <QBarLegendMarker>
#include <QBarSet>
#include <QLegend>
#include <QLegendMarker>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qbarlegendmarker.h>
#include "libqbarlegendmarker.h"
#include "libqbarlegendmarker.hxx"

QBarLegendMarker* QBarLegendMarker_new(QAbstractBarSeries* series, QBarSet* barset, QLegend* legend) {
	 return new VirtualQBarLegendMarker(series, barset, legend);
}

QBarLegendMarker* QBarLegendMarker_new2(QAbstractBarSeries* series, QBarSet* barset, QLegend* legend, QObject* parent) {
	 return new VirtualQBarLegendMarker(series, barset, legend, parent);
}

libqt_string QBarLegendMarker_Tr(const char* s) {
	QString _ret = QBarLegendMarker::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QBarLegendMarker_Type(QBarLegendMarker* self) {
	return self->type();
}

QAbstractBarSeries* QBarLegendMarker_Series(QBarLegendMarker* self) {
	return self->series();
}

QBarSet* QBarLegendMarker_Barset(QBarLegendMarker* self) {
	return self->barset();
}

libqt_string QBarLegendMarker_Tr2(const char* s, const char* c) {
	QString _ret = QBarLegendMarker::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QBarLegendMarker_Tr3(const char* s, const char* c, int n) {
	QString _ret = QBarLegendMarker::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QBarLegendMarker_QBaseType(QBarLegendMarker* self) {
	auto* vqbarlegendmarker = dynamic_cast<VirtualQBarLegendMarker*>(self);
	if (vqbarlegendmarker && vqbarlegendmarker->isVirtualQBarLegendMarker) {
vqbarlegendmarker->setQBarLegendMarker_Type_IsBase(true);
	return vqbarlegendmarker->type();
}
}

// Auxiliary method to allow providing re-implementation
void QBarLegendMarker_OnType(QBarLegendMarker* self, intptr_t slot) {
	auto* vqbarlegendmarker = dynamic_cast<VirtualQBarLegendMarker*>(self);
	if (vqbarlegendmarker && vqbarlegendmarker->isVirtualQBarLegendMarker) {
vqbarlegendmarker->setQBarLegendMarker_Type_Callback(reinterpret_cast<VirtualQBarLegendMarker::QBarLegendMarker_Type_Callback>(slot));
}
}

// Base class handler implementation
QAbstractBarSeries* QBarLegendMarker_QBaseSeries(QBarLegendMarker* self) {
	auto* vqbarlegendmarker = dynamic_cast<VirtualQBarLegendMarker*>(self);
	if (vqbarlegendmarker && vqbarlegendmarker->isVirtualQBarLegendMarker) {
vqbarlegendmarker->setQBarLegendMarker_Series_IsBase(true);
	return vqbarlegendmarker->series();
}
}

// Auxiliary method to allow providing re-implementation
void QBarLegendMarker_OnSeries(QBarLegendMarker* self, intptr_t slot) {
	auto* vqbarlegendmarker = dynamic_cast<VirtualQBarLegendMarker*>(self);
	if (vqbarlegendmarker && vqbarlegendmarker->isVirtualQBarLegendMarker) {
vqbarlegendmarker->setQBarLegendMarker_Series_Callback(reinterpret_cast<VirtualQBarLegendMarker::QBarLegendMarker_Series_Callback>(slot));
}
}

void QBarLegendMarker_Delete(QBarLegendMarker* self) {
    delete self;
}

